portal/views/portal.py

```diff
@@ -280,6 +302,10 @@ def access_via_token(token, next_step=None):
         vocabulary - see `NextStep`
 
     """
+    # Before going any further, confirm cookies are enabled
+    if not request.args.get('cookies_tested'):
+        return redirect(url_for('portal.require_cookies', target=request.url))
```

Comment:
Is there a way to implement this entirely in the frontend? I'd prefer that we don't add another redirection to the login process.

I understand that `/challenge` [will fail](https://github.com/uwcirg/true_nth_usa_portal/blob/develop/portal/views/portal.py#L437) before any frontend code can be loaded, because [the session cookie is never set, and unable to find the user](https://github.com/uwcirg/true_nth_usa_portal/blob/develop/portal/views/portal.py#L424). 
I think in that case, we should either redirect to an error page about cookies (similar to the `/no-script` page) and/or return a cookie-spefic error message than [what's returned](https://github.com/uwcirg/true_nth_usa_portal/blob/develop/portal/models/user.py#L1830).

@pbugni thoughts?

Reply:
hmm, this one is tough.  My understand from my discussion with Paul is that the originating call comes not from /challenge but /access - which makes a pure frontend implementation difficult.  Also, I think the error generated here - stemming from invalid user id - could come from reason other than browser cookie being disabled, e.g. session timed out, token expired, etc...  I however will defer to you and Paul on what the best approach here and will make the change accordingly.

Reply:
i only added this as i was under the impression that we could *not* implement this entirely in the frontend - @achen2401 didn't you experiment with this and prove (to yourself) that it required server side checks?

Reply:
>that it required server side checks?

I confirmed this too, roughly:

1. User clicks on email link (`/access`)
2. redirected to `/challenge`
3. [`/challenge` fails](https://github.com/uwcirg/true_nth_usa_portal/blob/develop/portal/views/portal.py#L437) because cookie never set during `/access`

I think a better solution would be to improve the error message, by checking explicitly checking for the session cookie

Reply:
think the request is aborted here:  https://github.com/uwcirg/true_nth_usa_portal/blob/c38fda20478fb75682552973941bcc35dab245b0/portal/views/portal.py#L435
so the response won't reach the frontend thereafter - I guess server side change is needed  there regardless for a pure frontend implementation.  I think that request could be aborted **for a number of reasons**, etc. no session cookie set, or bad token (token expired), which is why I thought it a good idea for redirecting for a frontend cookie check prior, so at least that will eliminate browser disabling of cookies as one of the reasons.  that is my thinking but am open to other implementations.

Reply:
>I think that request could be aborted for a number of reasons

Fair enough, what about redirecting only if there's a potential issue (ie `user_id is None` or `'session' not in request.cookies`), not every time?

>think the request is aborted here:

I think it's [actually here](https://github.com/uwcirg/true_nth_usa_portal/blob/c38fda20478fb75682552973941bcc35dab245b0/portal/views/portal.py#L437), the `POST` is for handling the validation of the challenge data AFAIK

>token expired

I think the error is `404 - Access token has expired` in that case